controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -93,6 +93,16 @@ public void processingFailure(Exception error) {
         return result;
     }
 
+    public CompletableFuture<Boolean> truncateSegment(final String scope,
+                                                    final String stream,
+                                                    final int segmentNumber,
+                                                    final long offset,
+                                                    final HostControllerStore hostControllerStore,
+                                                    final ConnectionFactory clientCF) {
+        // TODO: Issue #1965
```

Comment:
This should be throwing NotImplementedException or be implemented.

Reply:
If we throw notImplemented, then the workflow will fail. 
For now, the truncation workflow ignores this step and proceeds with deleting all segments that have been identified for deletion. 

Once the wireCommands for truncation are available, we will make the changes here to call it.

Reply:
if that comes, i can author an integration test to test it end to end as well.